js/zb.js

```diff
@@ -969,21 +969,117 @@ module.exports = class zb extends Exchange {
     }
 
     async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
-        if (type !== 'limit') {
-            throw new InvalidOrder (this.id + ' allows limit orders only');
-        }
         await this.loadMarkets ();
+        const market = this.market (symbol);
+        const swap = market['swap'];
+        const spot = market['spot'];
+        const action = params['action'];
```

Comment:
First, we can't refer to a key in a dictionary like `params['action']` cause that violates the following rules:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#accessing-dictionary-keys
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#sanitizing-input-with-safe-methods

So, if we check for a parameter in `params`, then we have to always do the following in line 976, just like you do in line 1000:

```JavaScript
const action = this.safeInteger (params, 'action');
```

However, the above isn't really necessary, cause if the user sets `params['action']` then it will override whatever action you set in the `request`, because each `request` is extended with `params` right before sending it out to the exchange, just like it is done in line 1053. So, if the user specified params['action'], that will end up in the `request` anyway. Hence, we don't need the if-else-if-else ladder and we don't need conditions like `(action === 1 || type === 1)` etc.

We could write the same logic in a shorter manner:

```JavaScript
            if (type === 'limit') {
                request['action'] = 1;
            } else if (timeInForce === 'IOC') {
                request['action'] = 3;
            } else if (timeInForce === 'PO') {
                request['action'] = 4;
            } else if (timeInForce === 'FOK') {
                request['action'] = 5;
            } else {
                request['action'] = type;
            }
            ...
        let response = await this[method] (this.extend (request, params));
```

The last `extend` call will propagate whatever action the user has set.